lib/libhoney/transmission.rb

```diff
@@ -136,7 +136,7 @@ def close(drain)
       end
 
       @batch_queue.enq(nil)
-      @batch_thread.join
+      @batch_thread.join unless @batch_thread.nil?
```

Comment:
This would also work. 

```suggestion
      @batch_thread.join if @batch_thread
```

I ... don't know which way I prefer. 🤔  I like `unless` until I don't.

Reply:
Hmm. Agreed about `unless` - I only ever use it in one-liners like this. I think my gut feel is that when checking for existence, I like to be as explicit as possible, so `unless @batch_thread.nil?` or `if @batch_thread.present?` would be my preference.

Reply:
🤔   You've convinced me with the "be explicit" reasoning. :shipit: 

![nodding in agreement](https://user-images.githubusercontent.com/517302/101390167-696b0700-3890-11eb-9366-a501a17c06a0.gif)